pkg/types/baremetal/validation/platform.go

```diff
@@ -290,6 +291,18 @@ func validateHostsCount(hosts []*baremetal.Host, installConfig *types.InstallCon
 	return nil
 }
 
+// ensure that the NetworkConfig field contains a valid Yaml string
+func validateNetworkConfig(hosts []*baremetal.Host, fldPath *field.Path) (errors field.ErrorList) {
+	for idx, host := range hosts {
+		networkConfig := make(map[interface{}]interface{})
+		err := yaml.Unmarshal([]byte(host.NetworkConfig), &networkConfig)
+		if err != nil {
+			errors = append(errors, field.Invalid(fldPath.Index(idx).Child("networkConfig"), host.NetworkConfig, "Not a valid yaml"))
+		}
```

Comment:
Suggestion: perhaps we can print the contents of `err` to gain more insight as to why the yaml is invalid?

Reply:
Thanks, done! The message is slightly too verbose/technical, but at least could give a better idea about the location of the error